spinbot.py

```diff
@@ -1,20 +1,67 @@
 #!/usr/bin/env python3
 
-import logging
-
 import event
 import gh
+import http
+import logging
+import os
 import policy
 import storage
+import sys
+import time
+
 from config import GetCtx
+from http.server import HTTPServer, BaseHTTPRequestHandler
+from io import BytesIO
+
+
+class SimpleServer(HTTPServer):
+    def __init__(self, server_address, handler):
+        super().__init__(server_address, handler)
+        self.ctx = GetCtx()
+        setup_logging(self.ctx)
+        setup_events(self.ctx)
+        setup_policies(self.ctx)
+
+        self.storage = create_storage(self.ctx)
+        self.github_client = create_client(self.ctx, self.storage)
+
+
+class SimpleHandler(BaseHTTPRequestHandler):
+
+    def do_GET(self):
+        self.send_response(200)
+        self.send_header('Content-type', 'text/plain')
+        self.end_headers()
+        b = bytes("working!", "UTF-8")
+        self.wfile.write(b)
+
+    def do_POST(self):
+        try:
```

Comment:
Do we really want long-running asynchronous batch processing to live in a synchronous web request?  While you did measure it and we're at ~50% of the default timeout for a web request, this still feels like something where we shouldn't leave a client waiting while we process; it feels like another failure mode that will break when we least expect it and will have to debug.

Reply:
Maybe even a minimal change here would be to return a 202 and asynchronously process?

Reply:
1. we're only at about 33% of the default timeout (106/300)
1. There should be no other clients beyond Cloud Scheduler, thus, returning a non 2xx error could then be monitored and alerted on with OOTB tools. A 202 response code would require someone to inspect logs to find any underlying issues.